testing/tests/test_charm_spec_autoload.py

```diff
@@ -126,6 +126,23 @@ def test_no_meta_raises(tmp_path, legacy):
             Context(charm)
 
 
+@pytest.mark.parametrize("legacy", (True, False))
+@pytest.mark.parametrize(
+    "params",
+    (
+        {"actions": {"foo": "bar"}},
+        {"config": {"foo": "bar"}},
+    ),
+)
+def test_partial_meta_raises(tmp_path, legacy, params):
+    with create_tempcharm(
+        tmp_path, legacy=legacy, meta={"type": "charm", "name": "sergio"}
+    ) as charm:
+        # metadata not found:
```

Comment:
I found this initially confusing because there's metadata two lines above, maybe this would be clearer?

```suggestion
        # metadata not found in YAML:
```